src/braket/default_simulator/densitymatrix_simulation_helpers.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2019-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
I'm not sure why we need a separate *helpers module. We should try and reduce the number of such "helpers" files. They tend to transform into a large collection of unrelated code over time, and don't end up being any help. I realize that we already have quite a few of these helpers in our packages, but we should try to avoid creating new ones if possible.